nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py

```diff
@@ -226,24 +226,67 @@ def __init__(
     ):
+        """
+        Load and prepare spike sorted data and corresponding metadata from the Neuroscope format (.res/.clu files).
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            Path to folder containing .clu and .res files.
+        gain : Optional[float], optional
+            Conversion factors from int16 to Volts are not contained in xml_file_path; set them explicitly here.
+            Most common value is 0.195 for an intan recording system.
+            The default is None.
+        xml_file_path : OptionalFilePathType, optional
+            Path to .xml file containing device and electrode configuration.
+            If unspecified, it will be automatically set as the only .xml file in the same folder as the .dat file.
+            The default is None.
+        keep_mua_units : bool
+            Optional. Whether or not to return sorted spikes from multi-unit activity.
+            The default is True.
+        exclude_shanks : list
+            Optional. List of indices to ignore. The set of all possible indices is chosen by default, extracted as the
+            final integer of all the .res.%i and .clu.%i pairs.
+        load_waveforms : bool, optional
+            If True, extracts waveform data from .spk.%i files in the path corresponding to
+            the .res.%i and .clue.%i files and sets these as unit spike features.
+            The default is False.
+            Not currently in use pending updates to NWB waveforms.
+        gain : float, optional
+            If loading waveforms, this value converts the data type of the waveforms to units of microvolts.
+            Conversion factors from int16 to Volts are not contained in xml_file_path; set them explicitly here.
+            Most common value is 0.195 for an intan recording system.
+            The default is None.
+            Not currently in use pending updates to NWB waveforms.
+        xml_file_path : PathType, optional
+            Path to the .xml file referenced by this sorting.
+        """
         assert HAVE_LXML, INSTALL_MESSAGE
 
         super().__init__(
             folder_path=folder_path,
             keep_mua_units=keep_mua_units,
             exclude_shanks=exclude_shanks,
-            load_waveforms=load_waveforms,
-            gain=gain,
+            # TODO: we can enable this once
+            #     a) waveforms on unit columns support conversion factor in NWB
+            #     b) write_sorting utils support writing said waveforms properly to a units table
+            # load_waveforms=load_waveforms,
+            # gain=gain,
         )
 
     def get_metadata(self):
         session_path = Path(self.source_data["folder_path"])
         session_id = session_path.stem
-        metadata = NeuroscopeRecordingInterface.get_ecephys_metadata(
+        metadata = dict()
+        metadata["Ecephys"] = NeuroscopeRecordingInterface.get_ecephys_metadata(
```

Comment:
shouldn't we be using the defined `xml_file_path` if it is provided?

Reply:
or is that done somewhere else?

Reply:
Oh, good catch! Fixing now.